iroha-lib/examples/DomainAssetCreation.cpp

```diff
@@ -0,0 +1,251 @@
+        return;
+    }
+
+    assert(response.has_account_assets_response());
+    auto accountAssetsResponce = response.account_assets_response();
+    for (const auto& r : response.account_assets_response().account_assets())
+    {
+        std::cout << "\tasset:" << r.asset_id() << " " << r.balance() << std::endl;
+    }
+
+    std::cout << LINE << "<" << __FUNCTION__ << LINE << std::endl;
+}
+
+iroha::protocol::Query generateGetAccountAssetsQuery(
+        const std::string& account_name,
+        const std::string& key_path)
+{
+    return generateQueryBase(account_name, key_path)
+            .getAccountAssets(account_name)
+            .signAndAddSignature();
+}
+
+iroha_lib::Query generateQueryBase(
+        const std::string& account_name,
+        const std::string& key_path)
+{
+    static auto log_manager = std::make_shared<logger::LoggerManagerTree>(
+                logger::LoggerConfig{logger::LogLevel::kInfo,
+                                     logger::getDefaultLogPatterns()})->getChild("CLI");
+
+    const auto keypair = generateKeypair(
+                account_name,
+                key_path,
+                log_manager);
+    return iroha_lib::Query(keypair);
+}
+
+iroha::protocol::Query generateGetAccountTransactionsQuery(
+        const std::string& account_name,
+        const std::string& key_path)
+{
+    return generateQueryBase(account_name, key_path)
+            .getAccountTransactions(account_name)
+            .signAndAddSignature();
+}
+
+iroha::protocol::Query generateGetAccountQuery(
+        const std::string& account_name,
+        const std::string& key_path)
+{
+    return generateQueryBase(account_name, key_path)
+            .getAccount(account_name)
+            .signAndAddSignature();
+}
+
+void printAccount(const std::string& account_name, const std::string& key_path, const std::string& peer_ip, uint16_t torii_port)
+{
+    std::cout << LINE << ">" << __FUNCTION__ << LINE << std::endl;
+
+    const auto query_proto = generateGetAccountQuery(
+                account_name,
+                key_path);
+    QueryResponse response = iroha_lib::GrpcClient(
+                peer_ip,
+                torii_port)
+            .send(query_proto);
+    auto payload = query_proto.payload();
+    assert(payload.get_account().account_id() == account_name);
+    assert(payload.has_get_account());
+
+    if (response.has_error_response())
+    {
+        const auto errorResponse = response.error_response();
+        std::cerr << errorResponse.error_code() << ": " << errorResponse.message() << std::endl;
+        return;
+    }
+
+    assert(response.has_account_response());
+    auto account = response.account_response().account();
+    std::cout << "account_id=" << account.account_id() << '\n'
+              << "domain_id=" << account.domain_id() << '\n'
+              << "quorum=" << account.quorum() << '\n'
+              << "json_data=" << account.json_data() << std::endl;
```

Comment:
Given that we use formatlib, can we consider using that here? `iostream` isn't particularly efficient, and IMO less clear than a formatlib statement would be.

Reply:
Yes, untying the C- and C++- io buffers is a good idea, given that we don't plan to link against any output producing C-code. 

The only reason not to use formatlib here (which is safer, faster and much more readable for people coming from Python and Rust), is if you don't want to link another dependency. But we already use it upstream, so why not use it here?